File: example-beego/models/response.go
package models

type Response struct {
	Code    int         `json:"code"`
	Message string      `json:"message"`
	Data    interface{} `json:"data"`
}

func MakeSuccess(data interface{}) Response {
	return Response{
		Code:    0,
		Message: "OK",
		Data:    data,
	}
}

func MakeFail(code int, msg string) Response {
	return Response{
		Code:    code,
		Message: msg,
	}
}

File: example-spring-boot/src/main/java/com/youthlin/example/boot/bean/Response.java
package com.youthlin.example.boot.bean;

/**
 * @author : youthlin.chen @ 2019-09-18 22:12
 */
public class Response<T> {
    private int code;
    private String msg;
    private T data;

    public int getCode() {
        return code;
    }

    public Response<T> setCode(int code) {
        this.code = code;
        return this;
    }

    public String getMsg() {
        return msg;
    }

    public Response<T> setMsg(String msg) {
        this.msg = msg;
        return this;
    }

    public T getData() {
        return data;
    }

    public Response<T> setData(T data) {
        this.data = data;
        return this;
    }
}
